internal/provider/cluster_resource.go

```diff
@@ -455,7 +469,7 @@ func (r *clusterResource) Update(ctx context.Context, req resource.UpdateRequest
 		clusterReq.SetDedicated(*dedicated)
 	}
 
-	clusterObj, _, err := r.provider.service.UpdateCluster(ctx, state.ID.ValueString(), clusterReq, &client.UpdateClusterOptions{})
+	clusterObj, _, err := r.provider.service.UpdateCluster(ctx, state.ID.ValueString(), clusterReq)
```

Comment:
The provider changes remove `UpdateClusterOptions` and `UpdateAllowlistEntryOptions` because they were removed in cockroach-cloud-sdk-go v0.4.0.